app/controllers/api/v2/instance_hosts_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Api
+  module V2
+    class InstanceHostsController < V2::BaseController
+      def resource_class
+        @resource_class ||= Host::Managed
+      end
+
+      def resource_scope(*args)
+        super.authorized(:view_hosts).joins(:infrastructure_facet).merge(::HostFacets::InfrastructureFacet.where(foreman: true))
+      end
+
+      api :PUT, '/instance/hosts/:host_id', N_("Assign a host to the Foreman instance")
+      param :host_id, :identifier_dottable
+      def update
+        # We cannot use resource scope as that is scoped only to hosts which already have the facet
+        host = ::Host::Managed.friendly.find(params[:id])
+        facet = host.infrastructure_facet || host.build_infrastructure_facet
+        facet.foreman = true
+        facet.save!
+        render status: :created, body: ''
+      end
+
+      api :DESTROY, '/instance/hosts/:host_id', N_("Unassign a given host from the Foreman instance")
```

Comment:
Looks like this returns a HTTP 204 No Content. Is that correct?

Reply:
What should it return? 200? And in that case, what should be in the body of the response?

Reply:
I think 204 is the right thing for a delete. And a 404 if it previously didn't exist can be an option.

Reply:
404 when it didn't exist before, same as https://github.com/theforeman/foreman/pull/8632#discussion_r701765102 . Do you insist on 204?

Reply:
Settled on 204 if the record previously existed, 404 if not